src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -93,10 +274,448 @@ struct boss_kazzara_the_hellforged : public BossAI
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_kazzara_energize::PeriodicTick, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 402547 - Hellbeam
+class spell_kazzara_hellbeam_consume_energy : public SpellScript
+{
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        if (GetHitUnit())
+            GetHitUnit()->SetPower(POWER_ENERGY, GetHitUnit()->GetPower(POWER_ENERGY) - 10);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_hellbeam_consume_energy::HandleHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 402260 - Ray of Anguish
+class spell_kazzara_ray_of_anguish_trigger : public SpellScript
+{
+    void HandleHitAura(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_RAY_OF_ANGUISH_AURA_45_SECONDS, true);
+    }
+
+    void HandleHitMissile(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_RAY_OF_ANGUISH_MISSILE, true);
+    }
+
+    void HandleHitAura2(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_RAY_OF_ANGUISH_AURA_10_SECONDS, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_ray_of_anguish_trigger::HandleHitAura, EFFECT_0, SPELL_EFFECT_DUMMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_ray_of_anguish_trigger::HandleHitMissile, EFFECT_1, SPELL_EFFECT_DUMMY);
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_ray_of_anguish_trigger::HandleHitAura2, EFFECT_2, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 404744 - Terror Claws
+class spell_kazzara_terror_claws : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
+        {
+            SPELL_TERROR_CLAWS_DAMAGE,
+            SPELL_TERROR_CLAWS_SHADOWFLAME_DAMAGE,
+            SPELL_TERROR_CLAWS_PERIODIC_AURA
+        });
+    }
+
+    void HandleHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        caster->CastSpell(GetHitUnit(), SPELL_TERROR_CLAWS_DAMAGE, true);
+        caster->CastSpell(GetHitUnit(), SPELL_TERROR_CLAWS_SHADOWFLAME_DAMAGE, true);
+        caster->CastSpell(GetHitUnit(), SPELL_TERROR_CLAWS_PERIODIC_AURA, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_kazzara_terror_claws::HandleHit, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 404743 - Terror Claws (periodic)
+class spell_kazzara_terror_claws_periodic : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_TERROR_CLAWS_PERIODIC_DAMAGE });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_TERROR_CLAWS_PERIODIC_DAMAGE, true);
```

Comment:
Are you sure the aura holder damages themselves? Or is it the aura owner 👀

Reply:
aura holder